### PR TITLE
Add set -e on top of each bash script in cookbooks

### PR DESCRIPTION
**Pull request by ddeidda**
This commit adds the missing "set -e" instructions to all bash scripts
to make them fail in case of errors.

### CODE REVIEW

File: recipes/tests.rb [hunk 1]
```diff
@@ -157,6 +159,7 @@
   bash 'test nvidia driver install' do
     cwd Chef::Config[:file_cache_path]
     code <<-TESTNVIDIA
+      set -e
```
Comment (lukeseawalker) [inline recipes/tests.rb]: must be not put here, it's few lines later

File: recipes/tests.rb [hunk 2]
```diff
@@ -183,6 +186,7 @@
   bash 'test CUDA install' do
     cwd Chef::Config[:file_cache_path]
     code <<-TESTCUDA
+      set -e
```
Comment (lukeseawalker) [inline recipes/tests.rb]: must be not put here, it's few lines later

File: recipes/tests.rb (hunk 1 above)
Comment (ddeidda) [reply recipes/tests.rb]: OK :+1:

File: recipes/tests.rb (hunk 2 above)
Comment (ddeidda) [reply recipes/tests.rb]: OK :+1:

File: recipes/tests.rb [hunk 3]
```diff
@@ -29,6 +29,7 @@
 bash 'check awscli regions' do
   cwd Chef::Config[:file_cache_path]
   code <<-AWSREGIONS
+    set -e
```
Comment (lukeseawalker) [inline recipes/tests.rb]: we could remove the exit 1

File: recipes/tests.rb [hunk 4]
```diff
@@ -139,6 +140,7 @@
   bash 'execute jq' do
     cwd Chef::Config[:file_cache_path]
     code <<-JQMERGE
+      set -e
```
Comment (lukeseawalker) [inline recipes/tests.rb]: we could remove the exit 1

File: recipes/tests.rb (hunk 3 above)
Comment (ddeidda) [reply recipes/tests.rb]: Done

File: recipes/tests.rb (hunk 4 above)
Comment (ddeidda) [reply recipes/tests.rb]: Done

**Pull request merged**